Java doc: "Configuring Ray" page (#10801)
ray-project/ray: doc/source/configure.rst

.. note:: Currently we don't provide a way to configure Ray when running a Java application in single machine mode. If you need to configure Ray, run ``ray start`` to start the Ray cluster first.

Driver Options
~~~~~~~~~~~~~~

There is a limited set of options for Java drivers. They are not for configuring the Ray cluster, but only for configuring the driver.

Ray uses `Typesafe Config <https://lightbend.github.io/config/>`__ to read options. There are several ways to set options:

- System properties. You can configure system properties either by adding options in the format of ``-Dkey=value`` in the driver command line, or by invoking ``System.setProperty("key", "value");`` before ``Ray.init()``.
- A `HOCON format <https://github.com/lightbend/config/blob/master/HOCON.md>`__ configuration file. By default, Ray will try to read the file named ``ray.conf`` in the root of the classpath. You can customize the location of the file by setting system property ``ray.config-file`` to the path of the file.

.. note:: Options configured by system properties have higher priority than options configured in the configuration file.

The list of available driver options:

- ``ray.address``

>   - The cluster address if the driver connects to an existing Ray cluster. If it's ``null``, a new Ray cluster will be created.

Reviewer: ```suggestion
  - The cluster address if the driver connects to an existing Ray cluster. If it is empty, a new Ray cluster will be created.
```

  - Type: ``String``

>   - Default: ``null``

Reviewer: ```suggestion
  - Default: empty string.
```
